File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.dart
class Solution {
  int minCostClimbingStairs(List<int> cost) {
    for (var i = 2; i < cost.length; i++) {
      cost[i] += min(cost[i - 1], cost[i - 2]);
    }
    return min(cost[cost.length - 1], cost[cost.length - 2]);
  }
}

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.rs
impl Solution {
    pub fn min_cost_climbing_stairs(mut cost: Vec<i32>) -> i32 {
        let n = cost.len();
        for i in 2..n {
            cost[i] += std::cmp::min(cost[i - 1], cost[i - 2]);
        }
        std::cmp::min(cost[n - 1], cost[n - 2])
    }
}
